appserver/data/data-core/src/main/java/fish/payara/jakarta/data/core/util/DeleteOperationUtility.java

```diff
@@ -58,6 +63,63 @@ public class DeleteOperationUtility {
 
     public static final Logger logger = Logger.getLogger(DeleteOperationUtility.class.getName());
 
+    public static int processDeleteByOperation(Object[] args, Class<?> declaredEntityClass, TransactionManager tm,
+                                               EntityManager em, EntityMetadata entityMetadata, Method method)
+            throws SystemException, NotSupportedException, HeuristicRollbackException,
+            HeuristicMixedException, RollbackException {
+
+        StringBuilder builder = new StringBuilder();
+        builder.append("DELETE FROM ").append(declaredEntityClass.getSimpleName())
+                .append(" o");
+
+        String attributeValue = null;
+        Annotation[][] parameterAnnotations = method.getParameterAnnotations();
+        int queryPosition = 1;
+        boolean hasWhere = false;
+
+        for (Annotation[] annotations : parameterAnnotations) {
+            for (Annotation annotation : annotations) {
+                if (annotation instanceof By) {
+                    attributeValue = ((By) annotation).value();
+                }
+                if (!hasWhere) {
+                    builder.append(" WHERE (");
+                    hasWhere = true;
+                } else {
+                    builder.append(" AND ");
+                }
+
+                if (attributeValue != null) {
+                    attributeValue = preprocessAttributeName(entityMetadata, attributeValue);
+                }
+                builder.append("o.").append(attributeValue).append("=?").append(queryPosition);
+            }
+            queryPosition++;
+        }
+
+        if (hasWhere) {
+            builder.append(")");
+        }
+
+        tm.begin();
+        em.joinTransaction();
+        Query q = em.createQuery(builder.toString());
+        for (int i = 0; i < args.length; i++) {
+            q.setParameter(i + 1, args[i]);
+        }
+        int rowsAffected = q.executeUpdate();
+        em.flush();
+        tm.commit();
+
+        logger.info("Rows affected from delete operation: " + rowsAffected);
+        return rowsAffected;
+    }
+
+    private static String preprocessAttributeName(EntityMetadata entityMetadata, String attributeName) {
+        // Add any necessary attribute name preprocessing logic here
```

Comment:
you need to validate against the metadata, here on the body of the method I can't see any comparation from the values added on the metadata fields for the entity, Could you check?

Reply:
you can refer to the method: FindOperationUtility.preprocessAttributeName(EntityMetadata entityMetadata, String attributeValue), the idea using this method on the find operation was to validate against the metadata from the entity, in case the value is not available an IllegalArgumentException wil be throw, please check that case

Reply:
You're right I completely forgot to implement this method